src/Temporalio/Client/EnvConfig/ClientEnvConfig.cs

```diff
@@ -0,0 +1,489 @@
+            /// Create a <see cref="TlsOptions"/> from this configuration.
+            /// </summary>
+            /// <returns>TLS options for a client.</returns>
+            public TlsOptions ToTlsOptions()
+            {
+                if (Disabled == true)
+                {
+                    return new TlsOptions();
+                }
+
+                return new TlsOptions
+                {
+                    Domain = ServerName,
+                    ServerRootCACert = ServerRootCACert?.Data,
+                    ClientCert = ClientCert?.Data,
+                    ClientPrivateKey = ClientPrivateKey?.Data,
+                };
+            }
+
+            /// <summary>
+            /// Convert to a record structure that can be used for TOML serialization.
+            /// </summary>
+            /// <returns>Record representation of this TLS config.</returns>
+            public TlsRecord ToRecord()
+            {
+                return new TlsRecord(
+                    Disabled: Disabled,
+                    ServerName: ServerName,
+                    ServerCaCert: DataSource.ToDictionary(ServerRootCACert),
+                    ClientCert: DataSource.ToDictionary(ClientCert),
+                    ClientKey: DataSource.ToDictionary(ClientPrivateKey));
+            }
+        }
+    }
+
+    /// <summary>
+    /// A data source for configuration, which can be a path to a file,
+    /// the string contents of a file, or raw bytes.
+    /// </summary>
+    public sealed class DataSource
+    {
+        private DataSource()
+        {
+        }
+
+        /// <summary>
+        /// Gets the file path for this data source, if applicable.
+        /// </summary>
+        public string? Path { get; private init; }
+
+        /// <summary>
+        /// Gets the raw data for this data source, if applicable.
+        /// </summary>
+        public byte[]? Data { get; private init; }
+
+        /// <summary>
+        /// Create a data source from a file path.
+        /// </summary>
+        /// <param name="path">Path to the configuration file.</param>
+        /// <returns>A new data source for the specified file.</returns>
+        /// <exception cref="ArgumentException">Thrown when path is null or empty.</exception>
+        public static DataSource FromPath(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentException("Path cannot be null or empty", nameof(path));
+            }
+
+            return new DataSource { Path = path };
+        }
+
+        /// <summary>
+        /// Create a data source from string content.
+        /// </summary>
+        /// <param name="content">Configuration data as a UTF-8 string.</param>
+        /// <returns>A new data source for the specified data.</returns>
+        /// <exception cref="ArgumentException">Thrown when content is null.</exception>
+        public static DataSource FromString(string content)
```

Comment:
```suggestion
        public static DataSource FromUTF8String(string content)
```

This would be a more .NET like name IMO